File: backend/blackbox/src/main/java/com/insight_web/blackbox/usecase/SessionInitializing.java
package com.insight_web.blackbox.usecase;

import com.insight_web.blackbox.adapters.auth.JwtUserDetailsService;
import com.insight_web.blackbox.adapters.auth.JwtUtils;
import com.insight_web.blackbox.domain.AuthToken;
import com.insight_web.blackbox.usecase.action.LoginAction;
import jakarta.transaction.Transactional;
import lombok.RequiredArgsConstructor;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class SessionInitializing {
    private final AuthenticationManager authenticationManager;
    private final JwtUtils jwtUtils;
    private final JwtUserDetailsService jwtUserDetailsService;

    @Transactional
    public AuthToken initialize(LoginAction action) {
        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(action.getEmail(), action.getPassword()));
        UserDetails userDetails = jwtUserDetailsService.loadUserByUsername(action.getEmail());
        var token = jwtUtils.generateToken(userDetails);
        return new AuthToken(token);
    }
}

File: backend/auth/src/main/kotlin/com/insightweb/auth/usecase/SessionInitializing.kt
package com.insightweb.auth.usecase

import com.insightweb.auth.adapters.auth.JwtTokenUtils
import com.insightweb.auth.adapters.auth.JwtUserDetailsHandler
import com.insightweb.auth.domain.AuthToken
import com.insightweb.auth.usecase.action.LoginAction
import jakarta.transaction.Transactional
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.stereotype.Service

@Service
class SessionInitializing(
    var authenticationManager: AuthenticationManager,
    var jwtUtils: JwtTokenUtils,
    var jwtUserDetailsHandler: JwtUserDetailsHandler,
) {
    @Transactional
    fun initialize(action: LoginAction): AuthToken {
        authenticationManager.authenticate(UsernamePasswordAuthenticationToken(action.email, action.password))
        val userDetails: UserDetails = jwtUserDetailsHandler.loadUserByUsername(action.email)
        val token = jwtUtils.generateToken(userDetails)
        return AuthToken(token)
    }
}